### PR TITLE
[XLA:CPU] Matmul + mul handling with OneDnn eltwise_linear.

### PR DESCRIPTION
**Pull request by jmaksymc**
This PR enables OneDNN call for the matched XLA HLO MatMul+Linear pattern using custom_call instruction. This PR:

* Enables MatMul+Linear pattern matching
* Using OneDNN MatMul post-ops handles MatMul+Linear fusion
* Adds a test for MatMul + Linear pattern

### CODE REVIEW

**Comment by kamaljeeti** [comment]
Hi @penpornk , can you look into this once? Thanks.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay!

File: xla/service/cpu/backend_config.proto [hunk 1]
```diff
@@ -26,9 +26,13 @@ message OneDnnMatMulConfig {
     GELU_ERF = 4;
     GELU_TANH = 5;
     BINARY_ADD = 6;
+    LINEAR = 7;
   }
   repeated FusionKind fused_ops = 3;
   bool bias_broadcast = 4;
+  // Because od proto failing for specific decimal values,
```
[the remark this answers is not in the document]
Comment (jmaksymc) [reply xla/service/cpu/backend_config.proto:33]: Hi, I'm currently on leave and will open an issue with an example when I'm back (01.30).

**Review by penpornk — APPROVED** [review]
Since the requested change is trivial and the author is OOO, I'll pull this in and make the changes internally.

File: xla/service/cpu/backend_config.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmaksymc) [reply xla/service/cpu/backend_config.proto:33]: @penpornk Here is an issue related to this problem: https://github.com/openxla/xla/issues/9010. Please let me know if you need more details.

**Pull request closed without merging**